server/src/citypay/citypay.service.ts

```diff
@@ -1,120 +1,135 @@
 
-      await page.goto(MS_APPLICANT_PORTAL.SIGN_IN_PAGE);
+    const { records: [firstPackage] } = await this.crmService.get('dcp_packages', `
+        $filter=dcp_packageid eq ${packageId}
+        &$expand=dcp_dcp_package_dcp_projectinvoice_package(
+          $filter=statuscode eq ${DCP_PROJECTINVOICE_CODES.statuscode.APPROVED}
+        )`
+    );
 
-      // this gets dynamically inserted after a handshake event with MS
-      await page.waitForSelector('[action="/SignIn"]');
+    const requestXML = await this.createRequestXML(firstPackage, agencyRequestID);
 
-      // enter credentials
-      await page.type('#Username', this.config.get('MS_APPLICANT_PORTAL_USERNAME'));
+    const params = new url.URLSearchParams({ saleData: requestXML });
 
-      // Something changed MCS-portal side for this selector
-      // the selector changed from Password -> PasswordValue
-      // this code will try both just in case.
-      try {
-        await page.type('#Password', this.config.get('MS_APPLICANT_PORTAL_PASSWORD'));
-      } catch (e) {
-        await page.type('#PasswordValue', this.config.get('MS_APPLICANT_PORTAL_PASSWORD'));
-      }
+    try {
+      let citypayResponse = null;
 
-      // click signin, but await for the response with cookies
-      page.click('#submit-signin-local');
+      citypayResponse = await axios.post(`${this.config.get('PAYMENT_BASE_URL')}/${this.config.get('PAYMENT_STEP1_URL')}`, params.toString(), {
+        headers: {
+          'content-type': 'application/x-www-form-urlencoded'
+        }
+      });
 
-      await page.waitForNavigation({
-        waitUntil: 'networkidle0'
+      const { data: xmlResponse } = citypayResponse;
+
+      const {
+        RetailPaymentResponse: {
+          receiptNumber: cartKey
+        }
+      } = create(xmlResponse).toObject() as {
+        RetailPaymentResponse: {
+          receiptNumber: string
+        }
+      };
+
+      const associatedInvoices = firstPackage.dcp_dcp_package_dcp_projectinvoice_package.map(projectInvoice => `/dcp_projectinvoices(${projectInvoice.dcp_projectinvoiceid})`)
+
+      // create new Project Invoice Postback in CRM
+      await this.invoicePostbackService.create({
+        dcp_name: agencyRequestID,
+        dcp_cartkey: cartKey,
+        dcp_postbackrequest: requestXML,
+        'dcp_dcp_projectinvoicepostback_dcp_projectinvoice_cartkey@odata.bind': associatedInvoices
       });
 
-      return page.cookies();
-    } catch (e) {
-      console.log(`having trouble... ${e}`);
-
-      // can't stop, won't stop...
-      // sometimes puppeteer times out.
-      if (retries) {
-        return this.stealCookies(retries - 1);
-      } else {
-        throw new Error(`Could not create cookie, maybe something is wrong with the username/password? ${e.toString()}`);
-      }
-    } finally {
-      const browser = await browserPromise;
-
-      browser.close();
+      return cartKey;
+    } catch(e) {
+      console.log("error: ", e);
```

Comment:
Not going to hold up this PR over this but think about how we can integrate Sentry into this work. I'm going to ask that we have any endpoints that we're rewriting hooked up to Sentry before we release this.